### DISCUSSION TITLE
What is the function of MachineBackground in the photonworkshop format?

### DISCUSSION THREAD

**Discussion by ejongejans** [sn4k3/UVtools#865] [General]
Hi Tiago, 

I'm investigating the MachineBackground property in the photonworkshop format. We had a user that is reporting thin film forming on the entire build platform. I noticed in the code base that this property was first 'unknown' and then received a name 'machinebackground'. 

I always assumed (after testing) that it had a value of 6506241 as it looked stable. 

I created files with anycubic workshop V322. The result was this: 

AnycubicPhotonD2.dl2p version: 517 machinebackgroundparameter:6506241
AnycubicPhotonM3Plus.pwmb version: 517 machinebackgroundparameter:6506241
AnycubicPhotonM3Premium.pm3r version: 517 machinebackgroundparameter:6506241
AnycubicPhotonMono.pwmo version: 515 machinebackgroundparameter:6506241
AnycubicPhotonMono2.pm3n version: 517 machinebackgroundparameter:6506241
AnycubicPhotonMonoM5.pm5 version: 517 machinebackgroundparameter:6506241
AnycubicPhotonMonoM5s.pm5s version: 518 machinebackgroundparameter:6506241
AnycubicPhotonMonoM5sPro.m5sp version: 518 machinebackgroundparameter:6506241
AnycubicPhotonMonoSE.pwms version: 515 machinebackgroundparameter:6506241
AnycubicPhotonMonoSQ.pmsq version: 515 machinebackgroundparameter:6506241
AnycubicPhotonMonoX6K.pwmb version: 517 machinebackgroundparameter:6506241
AnycubicPhotonMonoX6Ks.px6s version: 517 machinebackgroundparameter:6506241
AnycubicPhotonUltra.dlp version: 515 machinebackgroundparameter:6506241
AnycubicPhotonZero.pw0 version: 1 machinebackgroundparameter:6506241
AnycubicPhotonM3.pm3 version: 516 machinebackgroundparameter:**174278401**
AnycubicPhotonM3Max.pm3m version: 516 machinebackgroundparameter:**174278401**
AnycubicPhotonMono4k.pwma version: 516 machinebackgroundparameter:**191055617**
AnycubicPhotonMonoX.pwmx version: 516 machinebackgroundparameter:**962807553**
AnycubicPhotonMonoX2.pmx2 version: 517 machinebackgroundparameter:**996361985**

It got more strange when I loaded just 1 file, and set the machine to MonoX. 
Saving a sliced file repeatedly yields the following values for the machinebackground value: 

191055617
6506241
6506241
543377153
6506241
191055617

Any idea why you marked that machinebackground? 

kind regards
Elco

**Comment by sn4k3** [comment]
We are never sure for what that fields are or thier real meaning. 
From anycubic description and docs state that is the "machine background color", field is a char[4]. Whatever that means I don't know. But that property looks like info only.

For thin film, check if user have PerLayerSettings active, that flag causes problems with some anycubics: https://github.com/sn4k3/UVtools/issues/859

**Reply by ejongejans** [reply]
Ok thx. If you don't mind me asking, where do you get the anycubic description for these parameters? I could perhaps help testing every once in a while..

**Reply by sn4k3** [reply]
I were given a document(table) with most of the fields specs, still it does not enlighten in most cases. Many of them you still have to guess.

![image](https://github.com/sn4k3/UVtools/assets/113281/d7b8ecbb-8892-4ee0-8687-3d662c4e353d)

For example look at "property" the description ("property field of the machine") does not help at all in understanding it purpose.

All I know is also described here: https://github.com/sn4k3/UVtools/blob/master/Scripts/010%20Editor/PhotonWorkshop.bt
